metabase/metabase
# Migrate segment editor to MBQL lib

Epic https://github.com/metabase/metabase/issues/50336

Replaces the `GuiEditor` for segments with a new component based on MBQL lib + `EntityPicker` + `FilterPicker`. To keep this PR small I don't remove unused code here; also no new tests which I'll add separately. For end users the only noticeable change would be the `EntityPicker` instead of the old component.

How to verify:
- Admin settings -> Table metadata -> Segments
- Make sure you can create and edit segments

**Before**
<img width="1728" alt="Screenshot 2024-11-20 at 16 35 41" src="https://github.com/user-attachments/assets/f7b6fe43-a006-41f8-a171-eab497ed08d4">
<img width="1723" alt="Screenshot 2024-11-20 at 16 35 57" src="https://github.com/user-attachments/assets/5716b726-7a19-4d0e-8177-fa7dcef2cd46">

**After**
<img width="1728" alt="Screenshot 2024-11-21 at 12 53 49" src="https://github.com/user-attachments/assets/ec683767-fad4-4777-8257-65e6da91c6cb">
<img width="1728" alt="Screenshot 2024-11-21 at 12 36 58" src="https://github.com/user-attachments/assets/de62f9be-1965-40bd-82e9-8f3c50e65531">
<img width="1728" alt="Screenshot 2024-11-21 at 12 52 52" src="https://github.com/user-attachments/assets/4b19fa2a-d428-4c9d-83c5-bd07485b7c27">
<img width="951" alt="Screenshot 2024-11-21 at 12 36 29" src="https://github.com/user-attachments/assets/4d7588d9-9e59-4018-b216-d625065d12ed">
<img width="933" alt="Screenshot 2024-11-21 at 13 34 29" src="https://github.com/user-attachments/assets/239b8763-fa95-4ea9-bf38-4de8aecc1f1e">
